src/templates/base.html

```diff
@@ -33,204 +34,204 @@
-            var url = "{% url "competitions:detail" pk=0 %}".replace(0, pk)
-            return `{{ request.get_host }}${url}?secret_key=${secret_key}`
-        },
-
-        // Datasets
-        DATASET_MANAGEMENT: "{% url "datasets:management" %}",
-        DATASET_DOWNLOAD: function (key) {
-            return "{% url "datasets:download" key=0 %}".replace(0, key)
-        },
-
-        // Tasks
-        TASK_MANAGEMENT: "{% url "tasks:task_management" %}",
-        TASK_DETAIL: function(pk) {
-            return "{% url "tasks:detail" pk=0 %}".replace(0, pk)
+        // Load in URLs or other django generated goodies
+        URLS = {
+            assets: {
+                header_particles: "{% static "particles/header.json" %}"
+            },
+            API: "/api/",
+            STATIC: function (file_name) {
+                return "{% static '!-to-be-replaced-!' %}".replace('!-to-be-replaced-!', file_name)
+            },
+            // Websockets
+            //WS_SUBMISSION_LISTEN: ""
+
+            // Competitions
+            COMPETITION_MANAGEMENT: "{% url "competitions:management" %}",
+            COMPETITION_UPLOAD: "{% url "competitions:upload" %}",
+            COMPETITION_ADD: "{% url "competitions:create" %}",
+            COMPETITION_DETAIL: function (pk) {
+                return "{% url "competitions:detail" pk=0 %}".replace(0, pk)
+            },
+            COMPETITION_EDIT: function (pk) {
+                return "{% url "competitions:edit" pk=0 %}".replace(0, pk)
+            },
+            SECRET_KEY_URL: function (pk, secret_key) {
+                var url = "{% url "competitions:detail" pk=0 %}".replace(0, pk)
+                return `{{ request.get_host }}${url}?secret_key=${secret_key}`
+            },
+
+            // Datasets
+            DATASET_MANAGEMENT: "{% url "datasets:management" %}",
+            DATASET_DOWNLOAD: function (key) {
+                return "{% url "datasets:download" key=0 %}".replace(0, key)
+            },
+
+            // Tasks
+            // TODO:// update this URL when the task management view gets moved to the right place
+            TASK_MANAGEMENT: "{% url "tasks:task_management" %}",
+            TASK_DETAIL: function (pk) {
+                return "{% url "tasks:detail" pk=0 %}".replace(0, pk)
+            }
         }
-    }
-    // Django returns <input type="hidden" value="csrf-token-secret"> so we just grab the `value` attribute from this
-    CSRF_TOKEN = $('{% csrf_token %}').attr('value')
-</script>
-{% block extra_js %}
-{% endblock %}
-{% block extra_body %}
-{% endblock %}
+
+        // Django returns <input type="hidden" value="csrf-token-secret"> so we just grab the `value` attribute from this
+        CSRF_TOKEN = $('{% csrf_token %}').attr('value')
+
+        USER_LOGGED_IN = "{{ user.is_authenticated }}" === "True"
+        USER_IS_STAFF = "{{ user.is_staff }}" === "True"
+        USER_IS_SUPERUSER = "{{ user.is_superuser }}" === "True"
```

Comment:
Fixed in latest push and added a check to determine whether the user is logged in before assigning all these variables to prevent Javascript assignment errors.